src/test_workflow/integ_test/service.py

```diff
@@ -19,14 +20,30 @@ class Service(abc.ABC):
     Abstract base class for all types of test clusters.
     """
 
-    def __init__(self, work_dir, version, security_enabled, additional_config, dependency_installer):
+    def __init__(self, work_dir, filename, version, distribution, security_enabled, additional_config, dependency_installer):
+        self.filename = filename
         self.work_dir = work_dir
         self.version = version
+        self.distribution = distribution
         self.security_enabled = security_enabled
         self.additional_config = additional_config
         self.dependency_installer = dependency_installer
 
-        self.process_handler = Process()
+        self.process_handler = Process(self.filename, self.distribution)
```

Comment:
This class is a generic `Service`. Subclass it into `ServiceOpenSearchTar` and `ServiceOpenSearchRpm`, `ServiceOpenSearchDashboardsTar` etc.,  and implement the differences in the child classes.